internal/util/resource/resource_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"runtime"
+	"runtime/pprof"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type TestTrackObject struct {
+	token *Token
+}
+
+// runGC forces several GC cycles to give the runtime a chance to run cleanups.
+func runGC(t *testing.T) {
+	t.Helper()
+
+	for i := 0; i < 8; i++ {
+		runtime.GC()
+		time.Sleep(10 * time.Millisecond)
+	}
+}
+
+// entryCount returns the number of entries for obj in its pprof profile.
+func entryCount(t *testing.T, obj any) int {
+	t.Helper()
+
+	p := pprof.Lookup(profileName(obj))
+	if p != nil {
+		return p.Count()
+	}
+
+	return 0
+}
+
+func TestTrackProfileEntryAdded(t *testing.T) {
```

Comment:
It looks like many of those test cases could be merged into one

Reply:
Sure will do, generally I prefer to seperate tests to small parts that's why I did.